### PR TITLE
Add a test for Expression.Property and its handling of property accessors

### PR DESCRIPTION
**Pull request by vitek-karas**
This mostly a linker test as this effectively validates linker feature from https://github.com/mono/linker/pull/1880.

But having a clean trimming tests here is better validation for this specific case.

### CODE REVIEW

File: src/libraries/System.Linq.Expressions/tests/TrimmingTests/ExpressionPropertyTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            get => _testPropertyValue;
+            set { _testPropertyValue = value; }
+        }
+    }
+
+    static int Main(string[] args)
+    {
+        var obj = new TestType();
+        var param = Expression.Parameter(typeof(TestType));
+        var prop = Expression.Property(param, typeof(TestType).GetMethod("get_TestProperty"));
```
Comment (eerhardt) [inline src/libraries/System.Linq.Expressions/tests/TrimmingTests/ExpressionPropertyTests.cs:30]: Maybe it is overkill, but should we also have a test that has Roslyn creating the Expression? ex.

```C#
        var obj = new TestType();
        
        Expression<Func<TestType, bool>> expression = t => t.TestProperty;
        var prop = ((MemberExpression)expression.Body);
        ((PropertyInfo)prop.Member).SetValue(obj, true);

        if (obj._testPropertyValue != true)
            return -1;

        return 100;
```

**Review by eerhardt — APPROVED** [review]
Thanks for following up on this.

**Comment by eerhardt** [comment]
> But having a clean trimming tests here

Note: we don't verify our trimming tests don't have ILLink warnings.

https://github.com/dotnet/runtime/blob/ea69731ad1447345a01a4d918898dea8456bda68/eng/testing/linker/SupportFiles/Directory.Build.props#L6

**Comment by vitek-karas** [comment]
It would be nice to be able to validate that there were no warnings. I looking at that a little bit, but it's rather complicated... not a simple fix anywhere really.

File: src/libraries/System.Linq.Expressions/tests/TrimmingTests/ExpressionPropertyTests.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+{
+    static int Main(string[] args)
+    {
+        int result = ExplicitCreation.Test();
+        if (result != 100)
+            return result;
+
+        result = LambdaCreation.Test();
+
+        Type.GetType("foo");
```
Comment (eerhardt) [inline src/libraries/System.Linq.Expressions/tests/TrimmingTests/ExpressionPropertyTests.cs:24]: ??

Comment (vitek-karas) [reply src/libraries/System.Linq.Expressions/tests/TrimmingTests/ExpressionPropertyTests.cs:24]: I was trying to turn on the warnings - and this is an easy way to get a warning... but I left it in.

**Review by eerhardt — APPROVED** [review]

**Comment by eerhardt** [comment]
Linker tests passed. Runtime libraries test failures are unrelated.

**Pull request merged**